ddtrace/internal/sampling.py

```diff
@@ -82,3 +94,78 @@ def update_sampling_decision(
         return _set_trace_tag(context, sampling_mechanism)
     else:
         return _unset_trace_tag(context)
+
+
+class SpanSamplingRule:
```

Comment:
Should SpanSamplingRule be a child class of [BaseSampler]https://github.com/DataDog/dd-trace-py/blob/v1.2.2/ddtrace/sampler.py#L54?

```suggestion
class SpanSamplingRule(BaseSampler):
```

Reply:
I considered it but wasn't sure what we'd gain by doing so. Could you explain why we might want to?

Reply:
This would align with how define samplers. Actually Instead of subclassing `BaseSampler` it might make more sense to use `SamplingRule`. Functionality it wouldn't change much, so I have no strong opinions here

Reply:
This new rule class is internal so I do not see the benefit from sharing the base meta class.